packages/amplify-graphql-api-construct-tests/src/__tests__/migration/base-migration.test.ts

```diff
@@ -0,0 +1,157 @@
+import * as path from 'path';
+import {
+  createNewProjectDir,
+  deleteProjectDir,
+  deleteProject,
+} from 'amplify-category-api-e2e-core';
+import { initCDKProject, cdkDeploy, cdkDestroy, createGen1ProjectForMigration, writeTableMap, deleteDDBTables } from '../../commands';
+import { graphql } from '../../graphql-request';
+import { DURATION_20_MINUTES } from '../../utils/duration-constants';
+
+jest.setTimeout(DURATION_20_MINUTES);
+
+describe('Base Migration', () => {
+  let gen1ProjRoot: string;
+  let gen2ProjRoot: string;
+  let gen1ProjFolderName: string;
+  let gen2ProjFolderName: string;
+  let dataSourceMapping: Record<string, string>;
+
+  beforeEach(async () => {
+    gen1ProjFolderName = 'basemigrationgen1';
+    gen2ProjFolderName = 'basemigrationgen2';
+    gen1ProjRoot = await createNewProjectDir(gen1ProjFolderName);
+    gen2ProjRoot = await createNewProjectDir(gen2ProjFolderName);
+  });
+
+  afterEach(async () => {
+    try {
+      await deleteProject(gen1ProjRoot);
+    } catch (_) {
+      /* No-op */
+    }
+    try {
+      await cdkDestroy(gen2ProjRoot, '--all');
+    } catch (_) {
+      /* No-op */
+    }
+
+    try {
+      // Tables are set to retain when migrating from gen 1 to gen 2
+      await deleteDDBTables(Object.values(dataSourceMapping));
```

Comment:
If they're set to retain, why are we attempting to delete them? Should we be asserting that the delete fails? In fact, should we have an E2E that asserts that deletion protection is enabled for a migrating table?

Reply:
This is CFN deletion policy (not DDB deletion protection). The CFN deletion policy is set to retain on the Gen 1 and Gen 2 project.
1. The tables will still exist after the customer removes their gen 1 project.
2. The tables will still exist if the customer stops migrating and stays with gen 1.

The deletion is necessary to prevent resource leaks because the tables will not be removed after the `cdk destroy`/`amplify delete`.

The CFN deletion policy assertion is covered by the final list result result. This ensures that after the gen 1 project is deleted that the gen 2 project still has access to the DDB table. https://github.com/aws-amplify/amplify-category-api/pull/2703/files#diff-5085bba2562ae965a2878aeeae16f2cec073ff1564d2d4e459559c600fd42fd0R135-R137

I'll add some more comments so this is more clear.